src/Compilers/Core/Portable/Symbols/INamedTypeSymbolInternal.cs

```diff
@@ -16,5 +17,69 @@ internal interface INamedTypeSymbolInternal : ITypeSymbolInternal
 
         ImmutableArray<ISymbolInternal> GetMembers();
         ImmutableArray<ISymbolInternal> GetMembers(string name);
+
+        /// <summary>
+        /// True if this type or some containing type has type parameters.
+        /// </summary>
+        public bool IsGenericType { get; }
+
+        internal static class Helpers
+        {
+            /// <summary>
+            /// Returns True or False if we can determine whether the type is managed
+            /// without looking at its fields and Unknown otherwise.
+            /// Also returns whether or not the given type is generic.
+            /// </summary>
+            public static (ThreeState isManaged, bool hasGenerics) IsManagedTypeHelper(INamedTypeSymbolInternal type)
```

Comment:
>IsManagedTypeHelper [](http://example.com/codeflow?start=67&length=19)

Consider use a top-level class and make this an extension method.

Reply:
> Consider use a top-level class and make this an extension method.

I prefer to keep the code as is. I do not foresee us using this helper much in the future. Therefore, in my opinion there is no good reason to "pollute" name lookup binding space with it. If we could use Default Interface Implementations feature in our code, I would make this method a member of ```INamedTypeSymbolInternal```.